cmd/kcp-front-proxy/main.go

```diff
@@ -72,17 +78,33 @@ routed based on paths.`,
 			if errs := options.Validate(); errs != nil {
 				return errors.NewAggregate(errs)
 			}
-			proxyHandler, err := proxy.NewHandler(&options.Proxy)
+			var handler http.Handler
+			handler, err := proxy.NewHandler(&options.Proxy)
 			if err != nil {
 				return err
 			}
 
+			var authenticationInfo genericapiserver.AuthenticationInfo
 			var servingInfo *genericapiserver.SecureServingInfo
 			var loopbackClientConfig *restclient.Config
 			if err := options.SecureServing.ApplyTo(&servingInfo, &loopbackClientConfig); err != nil {
 				return err
 			}
-			doneCh, err := servingInfo.Serve(proxyHandler, time.Second*60, ctx.Done())
+			if err := options.Authentication.ApplyTo(&authenticationInfo, servingInfo); err != nil {
+				return err
+			}
+
+			// If an authenticator was set, wrap the handler with one that uses it
+			if authenticationInfo.Authenticator != nil {
+				scheme := runtime.NewScheme()
+				metav1.AddToGroupVersion(scheme, schema.GroupVersion{Group: "", Version: "v1"})
+				codecs := serializer.NewCodecFactory(scheme)
+
+				failedHandler := genericapifilters.Unauthorized(codecs)
```

Comment:
is the failedHandler called for everything other than client certs? That wouldn't make much sense.

Reply:
Yeah, I realized that and wrote a different handler.